java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -0,0 +1,412 @@
+                                       ColumnVector.BufferType buffer, long offset,
+                                       long length, byte[] tmpBuffer) throws IOException {
+    long left = length;
+    long at = offset;
+    while (left > 0) {
+      int amountToCopy = (int) Math.min(left, tmpBuffer.length);
+      column.copyHostBufferBytes(tmpBuffer, 0, buffer, at, amountToCopy);
+      out.write(tmpBuffer, 0, amountToCopy);
+      left -= amountToCopy;
+      at += amountToCopy;
+    }
+    return padFor64bitAlignment(out, length);
+  }
+
+  private static long copyBasicData(DataOutputStream out, ColumnVector column, long rowOffset,
+                                    long numRows, byte[] arrayBuffer) throws IOException {
+    DType type = column.getType();
+    long bytesToCopy = numRows * type.sizeInBytes;
+    long srcOffset = rowOffset * type.sizeInBytes;
+    return copyAndPad(out, column, ColumnVector.BufferType.DATA, srcOffset, bytesToCopy, arrayBuffer);
+  }
+
+  private static long copyStringData(DataOutputStream out, ColumnVector column, long rowOffset,
+                                     long numRows, byte[] arrayBuffer) throws IOException {
+    long startByteOffset = column.getStartStringOffset(rowOffset);
+    long endByteOffset = column.getEndStringOffset(rowOffset + numRows - 1);
+    long bytesToCopy = endByteOffset - startByteOffset;
+    long srcOffset = startByteOffset;
+    return copyAndPad(out, column, ColumnVector.BufferType.DATA, srcOffset, bytesToCopy, arrayBuffer);
+  }
+
+  private static long copyStringOffsets(DataOutputStream out, ColumnVector column, long rowOffset,
+                                        long numRows, byte[] arrayBuffer) throws IOException {
+    // If an offset is copied over as a part of a slice the first entry may be non-zero.  This is
+    // okay because we fix them up when they are deserialized
+    long bytesToCopy = (numRows + 1) * 4;
+    long srcOffset = rowOffset * 4;
+    return copyAndPad(out, column, ColumnVector.BufferType.OFFSET, srcOffset, bytesToCopy, arrayBuffer);
+  }
+
+  private static ColumnVector[] sliceOffColumnVectors(DType[] dataTypes, long[] nullCounts,
+                                                      TimeUnit[] timeUnits, long numRows,
+                                                      DeviceMemoryBuffer combinedBuffer,
+                                                      HostMemoryBuffer combinedBufferOnHost) {
+    int numColumns = dataTypes.length;
+    long combinedBufferOffset = 0;
+    ColumnVector[] vectors = new ColumnVector[numColumns];
+    boolean tableSuccess = false;
+    try {
+      for (int column = 0; column < numColumns; column++) {
+        DType type = dataTypes[column];
+        long nullCount = nullCounts[column];
+        TimeUnit tu = timeUnits[column];
+        DeviceMemoryBuffer validity = null;
+        DeviceMemoryBuffer data = null;
+        HostMemoryBuffer offsets = null;
+        boolean columnSuccess = false;
```

Comment:
Suggestion: Instead of tracking columnSuccess explicitly with a separate boolean and a nested try, could simplify this a bit by setting `validity`, `data`, and `offsets` to null after column successfully constructs then move the close checks for these to the `!tableSuccess` block.